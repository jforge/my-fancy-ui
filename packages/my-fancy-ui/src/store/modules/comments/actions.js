// @flow

import type {CommentType} from './types.js';

import {createAction} from 'redux-actions';
import {createActionTypeFactory} from 'redux-lumbergh';
import {moduleId, namepsace} from './config.js';

//
// Action types
//
const createActionType = createActionTypeFactory(namepsace, moduleId);

const actionTypes = {
  LOAD_COMMENTS: createActionType('LOAD_COMMENTS'),
  SET_COMMENTS: createActionType('SET_COMMENTS')
};

//
// Action creators
//
const loadComments = createAction(actionTypes.LOAD_COMMENTS);

export type SetCommentsPayloadType = {comments: Array<CommentType>};
const setComments = createAction(
  actionTypes.SET_COMMENTS,
  (comments: Array<CommentType> = []): SetCommentsPayloadType => ({
    comments
  })
);

const actions = {
  loadComments,
  setComments
};

export {actionTypes, actions};
