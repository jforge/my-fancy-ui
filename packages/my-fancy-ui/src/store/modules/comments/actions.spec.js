// @flow

import {actionTypes, actions} from './actions.js';

describe('actionTypes', () => {
  it('should be an object.', () => {
    expect(typeof actionTypes).toBe('object');
  });
});

describe('actions', () => {
  it('should be an object.', () => {
    expect(typeof actions).toBe('object');
  });

  it('actions.setComments() should create a payload containing the passed argument as the "comments" property', () => {
    // $FlowFixMe: suppressing this error since it is a test case
    const result = actions.setComments(['foo']);

    expect(result).toMatchSnapshot();
  });

  it('actions.setComments() should fallback to an empty array as the "comments" property if no argument was provided', () => {
    // $FlowFixMe: suppressing this error since it is a test case
    const result = actions.setComments();

    expect(result.payload.comments).toEqual([]);
  });

  it('actions.loadComments()', () => {
    const result = actions.loadComments();

    expect(result).toMatchSnapshot();
  });
});
