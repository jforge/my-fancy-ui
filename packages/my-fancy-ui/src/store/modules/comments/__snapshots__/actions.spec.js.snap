// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions actions.loadComments() 1`] = `
Object {
  "type": "@@immowelt/my-fancy-ui/comments/LOAD_COMMENTS",
}
`;

exports[`actions actions.setComments() should create a payload containing the passed argument as the "comments" property 1`] = `
Object {
  "payload": Object {
    "comments": Array [
      "foo",
    ],
  },
  "type": "@@immowelt/my-fancy-ui/comments/SET_COMMENTS",
}
`;
